core/src/smartcontracts/isi/world.rs

```diff
@@ -241,4 +338,15 @@ pub mod query {
             Ok(wsv.peers())
         }
     }
+
+    impl ValidQuery for FindAllPermissionTokenIds {
+        #[metrics("find_all_token_ids")]
+        fn execute(&self, wsv: &WorldStateView) -> Result<Self::Output, Error> {
+            Ok(wsv
+                .permission_token_ids()
+                .iter()
+                .map(|id_ref| id_ref.clone())
+                .collect())
```

Comment:
```suggestion
            Ok(wsv
                .permission_token_ids()
                .iter()
                .cloned()
                .collect())
```

Reply:
It's a [RefMulti](https://docs.rs/dashmap/5.3.4/dashmap/setref/multiple/struct.RefMulti.html), not an actual reference, so `cloned` doesn't work here.